### ISSUE TITLE
Updated to 2024.3.9 but the integration is now failing to start

### ISSUE THREAD

**Issue by mattenz** [ankohanse/hass-dab-pumps#23]
**Describe the bug**
After updating and then rebooting, the integration now has a "Setup failed, will retry" message. I've tried reinstalling the integration from HACS but that hasn't helped. It was working fine on 2024.3.8. In the log I see the following:

> Logger: custom_components.dabpumps.coordinator
> Source: custom_components/dabpumps/coordinator.py:308
> integration: DAB Pumps ([documentation](https://github.com/ankohanse/hass-dab-pumps), [issues](https://github.com/ankohanse/hass-dab-pumps/issues))
> First occurred: 1:34:35 PM (4 occurrences)
> Last logged: 1:34:59 PM
> 
> Retry 3 in 5 seconds. 'code'
> Retry 4 in 5 seconds. 'code'
> Retry 5 in 5 seconds. 'code'
> Retry 6 in 5 seconds. 'code'

**Comment by ankohanse** [comment]
Strange error, will need to dig a bit deeper to find what could have caused that, as it seems to run without issues for other users.
Is is possible to press the 'Download Diagnostics' button on the integration, or is it in a state where it doesn not show that button?

If you are able to download diagnostics, could you please email me the resulting file on 'info at anko dot co dot nz' ?

**Comment by mattenz** [comment]
Unfortunately, the "Download Diagnostics" option is missing. I did enable the debugging option though which shows the following repeat a few times, with this being the most recent:
```
2024-03-29 20:58:49.243 DEBUG (MainThread) [custom_components.dabpumps.api] DAB Pumps validate token via GET https://dconnect.dabpumps.com/api/v1/token/validatetoken
2024-03-29 20:58:49.692 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: dabpumps
2024-03-29 20:58:50.509 WARNING (MainThread) [custom_components.dabpumps.coordinator] Retry 7 in 5 seconds. 'code'
2024-03-29 20:58:52.174 DEBUG (MainThread) [custom_components.dabpumps.coordinator] Finished fetching DAB Pumps data in 60.001 seconds (success: False)
```

**Comment by mattenz** [comment]
As a test, I downgraded the integration back to 2024.3.8 and everything works fine again. My pump is a "EsyboxMini_V2_V1.0.0" on firmware 4.3.19 if that helps. I can upgrade again to help troubleshoot, tho :)

**Comment by ankohanse** [comment]
I found a possible (theoretical) spot that could lead to the trace "Retry x in 5 seconds. 'code'", but it indicates to me that the reponse returned from the DConnect servers during the login calls is not what I have seen so far.

Considering the setup of the integration has not reached the point where 'download diagnostics' becomes available, it would be really helpfull to figure what is going on if you would be willing to share your DAB Pumps login credentials with me. That way I can replay the calls to the DAB server in a debugger to see what is going on? And can test what alternatives there would be?
Don't post credentials in this issues report; instead email them to me on 'info at anko dot co dot nz'. I will treat them with the utmost confidentiality.

**Comment by mattenz** [comment]
I can do that. It's not really overly sensitive info :)

**Comment by ankohanse** [comment]
After some more communication with Matt, this issue should now be resolved in v2024.04.1
